app/views/spree/admin/orders/customer_details/edit.html.haml

```diff
@@ -9,7 +9,7 @@
 - content_for :page_actions do
   %li= button_link_to Spree.t(:back_to_orders_list), admin_orders_path, :icon => 'icon-arrow-left'
 
-- if @order.cart?
+- if @order.cart? || @order.state == 'address'
```

Comment:
ok, I investigated this a bit.
The edit action will always try to move the workflow forward.
In master the workflow is not moving from cart to address with error "email cant be empty".
In 3-0-latest the email validator is commented out so the workflow moves forward to address.
I think we can keep this change here @order.address? :-) because it makes sense in itself.
But we will need to address the problem with the email validator. Do you know if we have an issue for it @Matt-Yorkley ?
So, I'd say we can merge this PR but have an issue for the email validator.

Reply:
here's where the new email validator was introduced in spree:
https://github.com/openfoodfoundation/spree/commit/53972ecb3a44e177535b4d967fd910047fb7671c

Reply:
There's no matching helper for `.address?`

Reply:
Yes, there's an issue already.

Reply:
the state_machine always adds that helper for each state:
https://github.com/state-machines/state_machines#example

Reply:
ok, found it, linking to it here for reference: #4775

Reply:
PR updated to use cleaner `.address?` helper :+1: